test/Microsoft.Health.Fhir.Shared.Tests.Integration/Features/Operations/Export/CreateExportRequestHandlerTests.cs

```diff
@@ -171,6 +180,90 @@ public async Task GivenThereIsAMatchingJob_WhenRequestorClaimsInDifferentOrder_T
+        public async Task GivenARequestWithDifferentFormatNames_WhenConverted_ThenTheProperFormatStringIsReturned(string formatName, string expectedFormat)
+        {
+            _exportJobConfiguration.ExportJobFormats.Clear();
+            _exportJobConfiguration.ExportJobFormats.Add(new ExportJobFormatConfiguration()
+            {
+                Name = "test1",
+                Format = ExportFormatTags.ResourceName,
+            });
+            _exportJobConfiguration.ExportJobFormats.Add(new ExportJobFormatConfiguration()
+            {
+                Name = "test2",
+                Format = ExportFormatTags.Id,
+                Default = true,
+            });
+            _exportJobConfiguration.ExportJobFormats.Add(new ExportJobFormatConfiguration()
+            {
+                Name = "test3",
+                Format = ExportFormatTags.Timestamp,
+            });
+
+            ExportJobRecord actualRecord = null;
+            await _fhirOperationDataStore.CreateExportJobAsync(
+                Arg.Do<ExportJobRecord>(record =>
+            {
+                actualRecord = record;
+            }), Arg.Any<CancellationToken>());
+
+            var request = new CreateExportRequest(RequestUrl, ExportJobType.All, null, formatName: formatName);
+            CreateExportResponse response = await _createExportRequestHandler.Handle(request, _cancellationToken);
+
+            Assert.Equal(expectedFormat, actualRecord.ExportFormat);
+        }
+
+        [Theory]
+        [InlineData(false, ExportFormatTags.ResourceName)]
+        [InlineData(true, ExportFormatTags.Timestamp + "-" + ExportFormatTags.Id + "/" + ExportFormatTags.ResourceName)]
+        public async Task GivenARequest_WhenNoFormatsAreSet_ThenHardcodedDefaultIsReturned(bool containerSpecified, string expectedFormat)
+        {
+            _exportJobConfiguration.ExportJobFormats.Clear();
+
+            ExportJobRecord actualRecord = null;
+            await _fhirOperationDataStore.CreateExportJobAsync(
+                Arg.Do<ExportJobRecord>(record =>
+                {
+                    actualRecord = record;
+                }), Arg.Any<CancellationToken>());
+
+            var request = new CreateExportRequest(RequestUrl, ExportJobType.All, containerName: containerSpecified ? "test" : null);
+            CreateExportResponse response = await _createExportRequestHandler.Handle(request, _cancellationToken);
+
+            Assert.Equal(expectedFormat, actualRecord.ExportFormat);
+        }
+
+        [Fact]
+        public async Task GivenARequestWithAnNonexistantFormatName_WhenConverted_ThenABadRequestIsReturned()
+        {
+            var request = new CreateExportRequest(RequestUrl, ExportJobType.All, formatName: "invalid");
+            await Assert.ThrowsAsync<BadRequestException>(() => _createExportRequestHandler.Handle(request, _cancellationToken));
+        }
+
+        private T AddListener<T>(T baseObject)
```

Comment:
>AddListener [](start = 18, length = 11)

nit: I can checkout branch and put breakpoints inside Invoke method to find out what is going on here, but any chance some description can be added for this black magic reflection? Mainly what's it for?